### PR TITLE
Sync v2-9-stable with v2-9-test to release 2.9.0

### PR DESCRIPTION
**Pull request by ephraimbuddy**
Time for `2.9.0b2`!

### CODE REVIEW

File: RELEASE_NOTES.rst [hunk 1]
```diff
@@ -21,6 +21,285 @@
...
+
+The method ``get_permitted_menu_items`` in ``BaseAuthManager`` has been renamed ``filter_permitted_menu_items`` (#37627)
+""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+
+Add REST API actions to Audit Log events (#37734)
+"""""""""""""""""""""""""""""""""""""""""""""""""
+
+The Audit Log ``event`` name for REST API events will be prepended with ``api.`` or ``ui.``, depending on if it came from the Airflow UI or externally.
+
+Airflow 2.9.0 is the first release that officially supports Python 3.12 (#38025)
```
Comment (eladkal) [inline RELEASE_NOTES.rst]:
```suggestion
Official support for Python 3.12 (#38025)
```
We can shorten a bit :)

File: RELEASE_NOTES.rst [hunk 2]
```diff
@@ -21,6 +21,285 @@
...
+- Show dataset events above task/run details in grid view (#37603)
+- Introduce new config variable to control whether DAG processor outputs to stdout (#37439)
+- Make Datasets ``hashable`` (#37465)
+- Add conditional logic for dataset triggering (#37016)
+- Implement task duration page in react. (#35863)
+- Add ``queuedEvent`` endpoint to get/delete DatasetDagRunQueue (#37176)
+- Support multiple XCom output in the BaseOperator (#37297)
+- AIP-58: Add object storage backend for xcom (#37058)
+- Introduce ``DatasetOrTimeSchedule`` (#36710)
+- Add ``on_skipped_callback`` in to ``BaseOperator`` (#36374)
```
Comment (eladkal) [inline RELEASE_NOTES.rst]:
```suggestion
- Add ``on_skipped_callback`` to ``BaseOperator`` (#36374)
```

File: RELEASE_NOTES.rst [hunk 3]
```diff
@@ -21,6 +21,285 @@
...
+- Update faq.rst with (hopefully) clearer description of start_date (#36846)
+- Update public interface doc re operators (#36767)
+- Add ``exception`` to templates ref list (#36656)
+- Add auth manager interface as public interface (#36312)
+- Reference fab provider documentation in Airflow documentation (#36310)
+- Create auth manager documentation (#36211)
+- Update permission docs (#36120)
+- Docstring improvement to _covers_every_hour (#36081)
+- Add note that task instance, dag and lifecycle listeners are non-experimental (#36376)
+- Remove ``daskexcutor`` provider from Airflow code (#36269)
```
Comment (eladkal) [inline RELEASE_NOTES.rst]: This is provider only - we can skip it from airflow core change log

File: README.md [hunk 4]
```diff
@@ -98,9 +98,9 @@ Airflow is not a streaming solution, but it is often used to process real-time d
 
 Apache Airflow is tested with:
 
-|             | Main version (dev)         | Stable version (2.8.4)      |
+|             | Main version (dev)         | Stable version (2.9.0)      |
 |-------------|----------------------------|-----------------------------|
-| Python      | 3.8, 3.9, 3.10, 3.11, 3.12 | 3.8, 3.9, 3.10, 3.11        |
+| Python      | 3.8, 3.9, 3.10, 3.11, 3.12 | 3.8, 3.9, 3.10, 3.11, 3.12  |
 | Platform    | AMD64/ARM64(\*)            | AMD64/ARM64(\*)             |
 | Kubernetes  | 1.26, 1.27, 1.28, 1.29     | 1.25, 1.26, 1.27, 1.28, 1.29|
```
Comment (eladkal) [inline README.md]: i think we no longer support K8s 1.25 in Airflow 2.9?

File: RELEASE_NOTES.rst (hunk 3 above)
Comment (ephraimbuddy) [inline RELEASE_NOTES.rst]:
```suggestion
```

**Review by jedcunningham — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Review by eladkal — APPROVED** [review]

**Review by hussein-awala — APPROVED** [review]

**Pull request merged**